process explorer/backend/LocalCollector/Processes/ProcessInfo.cs

```diff
@@ -0,0 +1,227 @@
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Management;
+using System.Runtime.InteropServices;
+
+namespace ProcessExplorer.Entities
+{
+    public enum ProcessStatus
+    {
+        Running = 1,
+        Stopped = 0
+    }
+
+    public class ProcessInfo
+    {
+        public ProcessInfo(Process process)
+        {
+            //should refresh the process, because it can be already terminated
+            process.Refresh();
+            //process.HasExited and .StartTime has a bug due to the not having enough rights or they are elevated process, 
+            //and not a result of a process.Start call
+            if (process != null && process.Id != 0)
+            {
+                try
+                {
+                    StartTime = process.StartTime;
+                    ProcessorUsage = process.TotalProcessorTime;
+                    PhysicalMemoryUsageB = process.WorkingSet64;
+                    Status = process.HasExited == false ? ProcessStatus.Running : ProcessStatus.Stopped;
+                    UserProcessorTime = process.UserProcessorTime;
+                    ProcessPriorityClass = process.PriorityClass;
+                    VirtualMemorySize = process.VirtualMemorySize64;
+                    foreach (ProcessThread processThread in process.Threads)
+                    {
+                        Threads?.Append(processThread);
+                    }
+                }
+                catch (Exception)
+                {
+                    Status = ProcessStatus.Stopped;
+                }
+                finally
+                {
+                    ProcessId = process.Id;
+                    ProcessName = process.ProcessName;
+                    PriorityLevel = process.BasePriority;
+                    PagedMemoryUsage = process.PagedMemorySize64;
+                    NonPagedMemoryUsage = process.NonpagedSystemMemorySize64;
+                    PagedSystemMemoryUsage = process.PagedSystemMemorySize64;
+                    NonPagedSystemMemoryUsage = process.NonpagedSystemMemorySize64;
+                    PrivateMemoryUsage = process.PrivateMemorySize64;
+                    IsLinux = (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux));
+                    
+                    if (!IsLinux)
+                    {
+                        Children = GetChildProcessesWindows(process);
```

Comment:
This class is doing too much. You could use one abstraction to get certain attributes, and that one abstraction could have multiple - OS-specific - implementations